Its.Log.Monitoring/StatefulResultAssertions.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+
+namespace Its.Log.Monitoring
+{
+    public class StatefulResultAssertions<TState, TResult> where TResult : IComparable<TResult>
+    {
+        private readonly StatefulResult<TState, TResult> Subject;
+
+        public StatefulResultAssertions(StatefulResult<TState, TResult> statefulResult)
+        {
+            Subject = statefulResult;
+        }
+
+        public StatefulResultAssertions<TState, TResult> BeLessThan(TResult expected, string because = "")
+        {
+            var compareTo = Subject.Result.CompareTo(expected);
+            if (compareTo < 0)
```

Comment:
I see a bunch of places in this class where we can throw `NullReferenceException` if `Subject == null`. Is there a more meaningful result we can give back in that case?